specification/networkanalytics/resource-manager/Microsoft.NetworkAnalytics/stable/2023-11-15/networkanalytics.json

```diff
@@ -0,0 +1,2571 @@
+            "$ref": "./examples/DataTypes_GenerateStorageContainerSasToken_MaximumSet_Gen.json"
+          },
+          "DataTypes_GenerateStorageContainerSasToken_MaximumSet_Gen - generated by [MinimumSet] rule_MinimumSet_Gen": {
+            "$ref": "./examples/DataTypes_GenerateStorageContainerSasToken_MinimumSet_Gen.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.NetworkAnalytics/dataProducts/{dataProductName}/generateStorageAccountSasToken": {
+      "post": {
+        "operationId": "DataProducts_GenerateStorageAccountSasToken",
+        "tags": [
+          "DataProducts"
+        ],
+        "description": "Generate sas token for storage account.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "dataProductName",
+            "in": "path",
+            "description": "The data product resource name",
+            "required": true,
+            "type": "string",
+            "minLength": 3,
+            "maxLength": 63,
+            "pattern": "^[a-z][a-z0-9]*$"
+          },
+          {
+            "name": "body",
+            "in": "body",
+            "description": "The content of the action request",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/AccountSas"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "ARM operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/AccountSasToken"
+            }
+          },
+          "default": {
+            "description": "An unexpected error response.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "DataProducts_GenerateStorageAccountSasToken_MaximumSet_Gen": {
+            "$ref": "./examples/DataProducts_GenerateStorageAccountSasToken_MaximumSet_Gen.json"
+          },
+          "DataProducts_GenerateStorageAccountSasToken_MaximumSet_Gen - generated by [MinimumSet] rule_MinimumSet_Gen": {
+            "$ref": "./examples/DataProducts_GenerateStorageAccountSasToken_MinimumSet_Gen.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.NetworkAnalytics/dataProducts/{dataProductName}/listRolesAssignments": {
+      "post": {
```

Comment:
>post [](http://example.com/codeflow?start=7&length=4)

I forgot. Why isn't this list operation allowed by GET ting again? Don't you want resource readers to be able to see it?

Reply:
This is not the Top-level list response. This is a post call operation where we operate on a managed resource and respond back with list of roles.